src/create-text-tracks-if-necessary.js

```diff
@@ -16,14 +16,36 @@ const createTextTracksIfNecessary = function(sourceBuffer, mediaSource, segment)
 
   // create an in-band caption track if one is present in the segment
   if (segment.captions &&
-      segment.captions.length &&
-      !sourceBuffer.inbandTextTrack_) {
-    removeExistingTrack(player, 'captions', 'cc1');
-    sourceBuffer.inbandTextTrack_ = player.addRemoteTextTrack({
-      kind: 'captions',
-      label: 'cc1'
-    }, false).track;
-    player.tech_.trigger({type: 'usage', name: 'hls-608'});
+      segment.captions.length) {
+    if (!sourceBuffer.inbandTextTracks_) {
+      sourceBuffer.inbandTextTracks_ = {};
+    }
+
+    segment.captions.forEach(function(caption) {
```

Comment:
I'm not 100% sure I follow, so please do correct me if I've misunderstood. You're suggesting that the transmuxer keeps a local record of the tracks on which it's seen captions, and dispatches that information to media-sources? Like, say, when it dispatches a caption for a "new" track, it will set something like `caption.new = true`, which `media-sources` would then see and know it needs to create a new `remoteTextTrack` for that caption's `track` value?

That's certainly doable, and I agree that this loop on _every_ transmuxer push just feels wasteful. I'm not sure keeping state in the transmuxer is necessary, though. video.js historians might be more knowledgeable on this subject, but I'd hazard a guess that this code is only here anyway because the code in videojs/videojs-contrib-hls#1096 hasn't existed -- leaving this "track detection" code as the sole method for detecting and creating text tracks for CEA608 captions.

Perhaps some sort of logic could be included in videojs/videojs-contrib-hls#1096, something like, if closed captions are advertised in the playlist, then this code here would not be run, and we'd trust that all necessary caption tracks were declared in the playlist. If no text track exists, then we throw them into the void. And if captions aren't declared, but actually exist in the segments (I hope that is rare), then this loop would run.

That's potentially a breaking change from previous behaviour though, so please do let me know what you think.